### DISCUSSION TITLE
tonbo complete annotation

### DISCUSSION THREAD

**Discussion by caicancai** [tonbo-io/tonbo#157] [Ideas]
There is an obvious problem with tonbo at present, which is that the comments are not perfect. There are two disadvantages here. One is that it is not conducive to some developers to learn, such as those who want to use tonbo to get started with rust. The second is that tonbo actually lacks design documents. If the comments are not perfect, it is difficult for developers to understand why it is designed this way.

**Comment by caicancai** [comment]
@ethe what do you think

**Comment by ethe** [comment]
I agree with you, Tonbo does not have much documents and specifications in the current.
